Automattic/newspack-plugin
# feat(ia): display settings - part 1

### All Submissions:


### Changes proposed in this Pull Request:

- Add Recirculation section to Display Settings under Newspack > Settings.
- Add Author Bio section to Display Settings under Newspack > Settings.
- Enhanced `updateCacheKey` logic (inside `useWizardApiFetch`) to merge objects instead of replacing.
- Enhanced `THEME_BRAND_DEFAULTS` to include all known theme-mods. 
- Enhanced theme mods types to better structure and location.
- Refactored Theme and Brand (Typography, Theme Select), to incorporate `THEME_BRAND_DEFAULTS` changes.
- Refactored `checkbox` prop logic (inside WizardsActionCard) to consider `undefined` values.

### How to test the changes in this Pull Request:

1. Checkout this branch and build assets i.e. `git checkout origin/feat/ia-settings-display-settings && npm run build`
2. Navigate to _/wp-admin/admin.php?page=newspack-settings#/display-settings_
3. Observe Recirculation section has loaded. Test:
   - Updating settings
   - Saving settings
   - Data persists between page refreshes and tab transitions
![Screenshot 2024-09-13 at 12 07 23](https://github.com/user-attachments/assets/2f00bfa2-e8d3-45fb-9c07-4323e555fcaf)

4. Observe Author Bio section has loaded. Test:
   - Updating settings
   - Saving settings
   - Data persists between page refreshes and tab transitions
![Screenshot 2024-09-13 at 12 07 47](https://github.com/user-attachments/assets/f88e0d0a-c010-431d-923e-034e1cf5da41)

### Other information:
